palette/src/random_sampling/cone.rs

```diff
@@ -0,0 +1,173 @@
+use core::marker::PhantomData;
+
+use crate::float::Float;
+use crate::rgb::RgbSpace;
+use crate::{from_f64, FloatComponent, Hsl, Hsv};
+
+// Based on https://stackoverflow.com/q/4778147 and https://math.stackexchange.com/q/18686,
+// picking A = (0, 0), B = (0, 1), C = (1, 1) gives us:
+//
+// (  sqrt(r1) * r2  ,  sqrt(r1) * (1 - r2) + sqrt(r1) * r2  ) =
+// (  sqrt(r1) * r2  ,  sqrt(r1) - sqrt(r1) * r2 + sqrt(r1) * r2  ) =
+// (  sqrt(r1) * r2  ,  sqrt(r1)  )
+//
+// `sqrt(r1)` gives us the scale of the triangle, `r2` the radius.
+// Substituting, we get `x = scale * radius, y = scale` and thus for cone
+// sampling: `scale = powf(r1, 1.0/3.0)` and `radius = sqrt(r2)`.
+pub fn sample_cone<T: FloatComponent>(r1: T, r2: T) -> (T, T) {
+    let h = Float::cbrt(r1);
+    let r = h * Float::sqrt(r2);
```

Comment:
I'm going into some optimization possibilities here, but I think since this `h` is immediately divided away later, we can just remove this function completely and inline `r` and `h`.